src/renderers/shared/utils/ReactErrorUtils.js

```diff
@@ -155,84 +173,93 @@ if (__DEV__) {
       e,
       f,
     ) {
-      depth++;
-
-      let error;
+      // Keeps track of whether the user-provided callback threw an error. We
+      // set this to true at the beginning, then set it to false right after
+      // calling the function. If the function errors, `didError` will never be
+      // set to false. This strategy works even if the browser is flaky and
+      // fails to call our global error handler, because it doesn't rely on
+      // the error event at all.
       let didError = true;
+
+      // Create an event handler for our fake event. We will synchronously
+      // dispatch our fake event using `dispatchEvent`. Inside the handler, we
+      // call the user-provided callback.
       const funcArgs = Array.prototype.slice.call(arguments, 3);
-      const boundFunc = function() {
+      const onFakeEvent = function() {
```

Comment:
Let's be honest, this is going to look confusing in a stack trace regardless :D I'll change it to something more specific, though.